apps/web/src/components/Publication/PublicationBody.tsx

```diff
@@ -36,16 +37,19 @@ const PublicationBody: FC<PublicationBodyProps> = ({
     'quotedPublicationId'
   );
 
-  let content = metadata?.content;
   const filterId = snapshotProposalId || quotedPublicationId;
 
-  if (filterId) {
-    for (const url of urls) {
-      if (url.includes(filterId)) {
-        content = content?.replace(url, '');
+  const [content, setContent] = useState(metadata?.content);
+
+  useEffect(() => {
```

Comment:
useEffect is not suggested here 🙈 

https://overreacted.io/a-complete-guide-to-useeffect/